### ISSUE TITLE
Keep getting error after update to 0.3.3

### ISSUE THREAD

**Issue by Gozala** [ipfs/go-ipfs#1213]
```sh
Received interrupt signal, shutting down...
Publishing!
17:39:32.927 ERROR     ipnsfs: leveldb: closed system.go:83
```

**Comment by Gozala** [comment]
Ok I noticed following in my config:

```
"Datastore": {
    "Type": "leveldb",
    "Path": "/Users/gozala/.go-ipfs/datastore"
  },
```

And there is no `.go-ipfs` directory on my machine instead there is `.ipfs` which is probably why I get that error. Not sure if that was a migration that screw up a config or what but I think this is a reason I wind up with broken `ipfs`.

**Comment by Gozala** [comment]
Ok it looks like wrong path was not an issue, as fixing a config does not changed anything.

**Comment by whyrusleeping** [comment]
Hrm, this appears to be a leftover from a time when our datastores were actually configuruable [insert whyrusleeping's grumblings about the fsrepo here].

We should revert to respecting this.

**Comment by Gozala** [comment]
@whyrusleeping in fact I have completely removed `.ipfs` & inited it from scratch but I still seem to get same error. In fact very few things work after update to `ipfs@0.3.3` could be that this issue somehow related to #1214

**Comment by whyrusleeping** [comment]
oh, the error log you are seeing isnt actually a problem, its just a reminder for us to fix the daemon shutdown code.

**Comment by jbenet** [comment]
@Gozala thanks for bringing this up. as @whyrusleeping mentions, this is a confluence of a few things. I think the main problem you're seeing is the resolver problem #1212 addressed -- which definitely would break many things. (Sorry about the confusion-- the tests and I both missed the bug when we merged it in yesterday.)

---

> Ok it looks like wrong path was not an issue, as fixing a config does not changed anything.

Yeah, this brings up another cleanup bug: we should remove those keys from the config as they're not used anymore the way they used to be. (we should definitely remove `Datastore.Path` and maybe `Datastore.Type` too, as `leveldb/flatfs` mix is not optional at the moment.

Or maybe keep `Datastore.Type` until we figure out how we'll distinguish storage once the `s3 datastore` (and possible full `s3 repo`) come around. (cc @tv42)

---

@whyrusleeping we should either make
```
Publishing!
```
more useful to the user, or remove it. if keeping, maybe something like:
```
auto-publishing mounted /ipns/<name> to /ipfs/<path>
```

---

summary:
- [x] merged #1212 (should fix main problem)
- [ ] test whether #1212 merged fixes your error (@Gozala)
- [ ] remove `Datastore.path` from config, as it is no longer path-configurable that way. (probably worth adding to a next migration)
- [ ] perhaps remove `Datastore.Type` as that is not configurable either anymore (we should have  removed support for that cleanly, or not at all...)
- [ ] make `Publishing` more informative or remove it. (@whyrusleeping ?)

**Comment by jbenet** [comment]
thanks for bringing this up @Gozala ! sorry for confusing error cases atm. this managed to hit lots of misleading red herrings. :/

**Comment by Gozala** [comment]
@jbenet sure np, yeah no worries I think combination of this errors and webui not working just made it hard to guess what's going on ;)

**Comment by jbenet** [comment]
@Gozala was the error you were seeing fixed?

**Comment by Gozala** [comment]
Jus tried version from master, still seem to not work for me (posted output of errors I the other thread)
https://github.com/ipfs/go-ipfs/issues/1214#issuecomment-101034213

**Comment by diasdavid** [comment]
IPNS and files (old ipnsfs or mfs) had a lot of improvements in the last months, specially in the close to be released 0.4.0 (check dev0.4.0 branch). @Gozala would you mind testing it again and see if the error persists?

**Comment by whyrusleeping** [comment]
I beleive this has since been resolved (or is no longer an issue). Feel free to open a new issue if there are still problems.
